core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/TxnStatusFile.java

```diff
@@ -13,41 +13,70 @@
 
 import org.eclipse.rdf4j.common.io.NioFile;
 
+import static java.nio.charset.StandardCharsets.US_ASCII;
+
 /**
  * Writes transaction statuses to a file.
  */
 class TxnStatusFile {
+	public static final byte NONE_BYTE = (byte) 0b00000001;
+	public static final byte ACTIVE_BYTE = (byte) 0b00000010;
+	public static final byte COMMITTING_BYTE = (byte) 0b00000100;
+	public static final byte ROLLING_BACK_BYTE = (byte) 0b00001000;
+	public static final byte UNKNOWN_BYTE = (byte) 0b00010000;
 
 	public static enum TxnStatus {
 
 		/**
 		 * No active transaction. This occurs if no transaction has been started yet, or if all transactions have been
 		 * committed or rolled back.
 		 */
-		NONE,
+		NONE(NONE_BYTE),
 
 		/**
 		 * A transaction has been started, but was not yet committed or rolled back.
 		 */
-		ACTIVE,
+		ACTIVE(ACTIVE_BYTE),
 
 		/**
 		 * A transaction is being committed.
 		 */
-		COMMITTING,
+		COMMITTING(COMMITTING_BYTE),
 
 		/**
 		 * A transaction is being rolled back.
 		 */
-		ROLLING_BACK,
+		ROLLING_BACK(ROLLING_BACK_BYTE),
 
 		/**
 		 * The transaction status is unknown.
 		 */
-		UNKNOWN;
+		UNKNOWN(UNKNOWN_BYTE);
+
+		byte[] onDisk;
```

Comment:
Can we make this a private (final) field with a separate accessor method?

Reply:
Done.